core/src/test/scala/dagr/core/config/ConfigurationTest.scala

```diff
@@ -102,9 +105,24 @@ class ConfigurationTest extends UnitSpec {
 
   it should "find executables" in {
     conf.configureExecutable("some-executable", "n/a") shouldBe Paths.get("/does/not/exist")
+    conf.configureExecutableFromBinDirectory("some-executable", "exec") shouldBe Paths.get("/does/not/exist/exec")
 
-    val java = conf.configureExecutable("java.exe", "java")
+    var java = conf.configureExecutable("java.exe", "java")
     java.getFileName.toString shouldBe "java"
     Files.isExecutable(java) shouldBe true
+    java = conf.configureExecutableFromBinDirectory("path-does-not-exist", "java")
```

Comment:
Can you add a comment here that, even though you're providing a bin directory, you still expect this to fall back to the system path and thus succeed?